### DISCUSSION TITLE
Migrate TextExpander to Espanso

### DISCUSSION THREAD

**Discussion by leingang** [espanso/espanso#1476] [Q&A]
I am a long-time [TextExpander](https://textexpander.com/) for Mac user, but I've started playing with Espanso and like what I see. I was wondering if anybody has already attempted to migrate TextExpander's snippets to Espanso.

In theory, it should be do-able, since TextExpander stores snippets as Apple property list files. I don't think the spec is open but it can be reverse-engineered by reading the files. So it's a classic data munging problem. With the right python library's and a free Saturday afternoon I think I could get it 80% done (the ones which are straight-up substitutions). But I thought I'd ask in case somebody's already done it.

**Comment by smeech** [comment]
Sorry about the late reply - I am trying to mop up and sort the old Issues and Discussions.

I've read extensively about Espanso in recent weeks but haven't come across anyone else's work in this area.

There's a neat online [AHK to Espanso converter](https://ahk-to-espanso.as93.net/)!
